src/wp-includes/blocks/image.php

```diff
@@ -185,22 +185,37 @@ function block_core_image_render_lightbox( $block_content, $block ) {
 	$p->seek( 'figure' );
 	$figure_class_names = $p->get_attribute( 'class' );
 	$figure_styles      = $p->get_attribute( 'style' );
+
+	// Create unique id and set the image metadata in the state.
+	$unique_image_id = uniqid();
```

Comment:
Should this be one of these to match the other blocks?

* `wp_generate_uuid4()` (if stored in DB)
* `wp_unique_id( 'some-prefix' )`
* `wp_unique_prefixed_id( 'some-prefix' )`

Reply:
A related comment from @luisherranz is here: https://github.com/WordPress/gutenberg/pull/63348#discussion_r1675709371. 

I see the [first iteration of the PR](https://github.com/WordPress/gutenberg/pull/63348/commits/b1e8c2c7f9c7a42fb381a47e01bcbc1338cd1c67) had something similar to the options listed by you:

```php
// Create unique id and set the image metadata in the state.
$unique_image_id = wp_unique_id( 'image-' );
```

It doesn't guarantee that ids are unique between several pages.

Reply:
That's right, with client-side navigation we need to ensure that those IDs are unique not only within the same page render but across different page renderings as well.